Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js

```diff
@@ -369,13 +369,33 @@ WI.JavaScriptLogViewController = class JavaScriptLogViewController extends WI.Ob
                 var groupElement = group.render(messageView);
                 this._currentSessionOrGroup.append(groupElement);
                 this._currentSessionOrGroup = group;
-            } else
-                this._currentSessionOrGroup.addMessageView(messageView);
+            } else this._currentSessionOrGroup.addMessageView(messageView);
+            
+            if (WI.settings.experimentalGroupSourceMapErrors.value && WI.consoleManager.failedSourceMapConsoleMessages.has(messageView.message))
+                this._addConsoleMessageToConsoleSourceMapErrorGroup(messageView);
```

Comment:
it's a bit odd that we do this after the above logic as i think that means the `messageView` gets `append` twice, once to the `_currentSessionOrGroup` and again to the `this._failedSourceMapsGroup`.  is it necessary for us to do that twice or could we move this check to be the primary `if` condition (with the existing `if` becoming `else if`)?

Reply:
This is an intentional decision to avoid the `ConsoleSourceMapMessageGroup` from being added to an opened ConsoleGroup. As a special case of grouped errors, we want to ensure it is always top-level and not unintentionally hidden by being nested in a collapsed group whether created by us or by the user. cc @rcaliman-apple

Reply:
it's probably not a big deal since it only applies to sourcemap errors (which are likely few/none of all the console messages), but it would be nice for this to not be a concern at all in the first place

i think we could have `_addConsoleMessageToConsoleSourceMapErrorGroup` return `true`/`false` based on whether it "consumed" the `messageView` (which is based on whether it's the first sourcemap error) and then if `true` it can early return at this callsite

im fine with this being a followup if you'd prefer :)

Reply:
Sounds like a plan.

Reply:
Filed https://webkit.org/b/264490

Reply:
can you please put a `// FIXME: <https://webkit.org/b/264490> (Improve failed SourceMap error grouping to avoid unnecessary work)` comment here so there's better linkage between bugzilla and the source code?